src/sqlfluff/dialects/dialect_exasol.py

```diff
@@ -3759,7 +3759,7 @@ class FunctionScriptStatementSegment(BaseSegment):
 
 
 @exasol_dialect.segment(replace=True)
-class StatementSegment(BaseSegment):
+class StatementSegment(ansi_dialect.get_segment("StatementSegment")):  # type: ignore
```

Comment:
Is it better to do this? Or just add the `get_table_references` function to this Segment instead? We normally do the latter.

Reply:
If the code is the same, I'd say this is better. Code reuse is one of the main benefits of inheritance, and copy paste. Especially in this case, where we'd be copying a good bit of code (about 10 lines).

Reply:
Oh definitely wasn’t suggesting copying and pasting the whole function. It’s whether you inherit the class or point the function to the ansi one. We tend to do the latter. But maybe that’s not the best way and this is better.

Reply:
Got it. I don't have a strong feeling. One reason I did it this way is that it always works, even if the base class had several functions -- inheriting gives you access to them all, while the other way might require you to copy individual things, one by one.

TBH, I think it'd be good if we had a cleaner way to do this. Currently, the code is ugly either way. We could probably support something like this pretty easily:
```
class StatementSegment(ansi_dialect.StatementSegment)
```

That might allow us to omit the `# type: ignore` comments too.

Reply:
Yeah was curious why we needed to ignore the type here?

Reply:
The type checker sees the function call `get_segment()` being used to specify the base class for the class we're creating, and it thinks, "oh, this is tricky. That function could return anything, so I can't analyze this the usual way, where I assume the base class is a known, specific thing." In reality, `get_segment()` is predictable, too, but the type checker doesn't know about it. I'm thinking if we changed this so we were using `.`, the type checker would assume it's more predictable. (It's not really!)